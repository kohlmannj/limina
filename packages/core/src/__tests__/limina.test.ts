import { createBreakpointValueConstructors, limina, createBreakpoints } from '..';

/** @see https://codepen.io/anon/pen/VXJgom */
/** @see https://codesandbox.io/s/jj438z6r2w */
/** @see https://codesandbox.io/s/x256m5vl54 */

describe('limina', () => {
  it('basic usage', () => {
    const [compactBreakpoint, regularBreakpoint] = createBreakpoints(540, 1280);

    const [compact, regular] = createBreakpointValueConstructors(540, 1280);

    const styles = limina({
      width: [compact(540), regular(1160)],
    });

    expect(styles).toMatchSnapshot();
  });
  describe('styles object with array of BreakpointValues', () => {
    describe('two breakpoints', () => {
      it('works with a single property-value pair', () => {
        const [compact, regular] = createBreakpointValueConstructors(540, 1280);

        const styles = limina({
          // TODO: order values before calling retargetCSSPropertyValue()
          width: [compact(540), regular(1160)],
        });

        expect(styles).toMatchSnapshot();
      });
    });

    describe('three breakpoints', () => {
      it('works with a single property-value pair', () => {
        const [mobile, tablet, desktop] = createBreakpointValueConstructors(540, 854, 1280);

        const styles = limina({
          width: [mobile(540), tablet(768), desktop(1160)],
        });

        expect(styles).toMatchSnapshot();
      });

      it('works with a multiple property-value pairs', () => {
        const [mobile, tablet, desktop] = createBreakpointValueConstructors(540, 854, 1280);

        const styles = limina({
          width: [mobile(540), tablet(768), desktop(1160)],
          fontSize: [mobile(32), tablet(48), desktop(64)],
        });

        expect(styles).toMatchSnapshot();
      });
    });
  });
});
