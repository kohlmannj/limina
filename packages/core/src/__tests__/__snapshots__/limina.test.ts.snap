// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`limina basic usage 1`] = `
Object {
  "@media (max-width: 540px)": Object {
    "width": "100vw",
  },
  "@media (min-width: 1280px)": Object {
    "width": "90.625vw",
  },
  "@media (min-width: 541px) and (max-width: 1279px)": Object {
    "width": "calc(83.78378378378379vw + 87.56756756756761px)",
  },
}
`;

exports[`limina styles object with array of BreakpointValues three breakpoints works with a multiple property-value pairs 1`] = `
Object {
  "@media (max-width: 540px)": Object {
    "fontSize": "5.9259259259259265vw",
    "width": "100vw",
  },
  "@media (min-width: 1280px)": Object {
    "fontSize": "5vw",
    "width": "90.625vw",
  },
  "@media (min-width: 541px) and (max-width: 853px)": Object {
    "fontSize": "calc(5.095541401273886vw + 4.484076433121018px)",
    "width": "calc(72.61146496815286vw + 147.89808917197456px)",
  },
  "@media (min-width: 854px)": Object {
    "fontSize": "5.620608899297424vw",
    "width": "89.92974238875878vw",
  },
  "@media (min-width: 855px) and (max-width: 1279px)": Object {
    "fontSize": "calc(3.755868544600939vw + 15.924882629107984px)",
    "width": "calc(92.01877934272301vw + -17.840375586854407px)",
  },
}
`;

exports[`limina styles object with array of BreakpointValues three breakpoints works with a single property-value pair 1`] = `
Object {
  "@media (max-width: 540px)": Object {
    "width": "100vw",
  },
  "@media (min-width: 1280px)": Object {
    "width": "90.625vw",
  },
  "@media (min-width: 541px) and (max-width: 853px)": Object {
    "width": "calc(72.61146496815286vw + 147.89808917197456px)",
  },
  "@media (min-width: 854px)": Object {
    "width": "89.92974238875878vw",
  },
  "@media (min-width: 855px) and (max-width: 1279px)": Object {
    "width": "calc(92.01877934272301vw + -17.840375586854407px)",
  },
}
`;

exports[`limina styles object with array of BreakpointValues two breakpoints works with a single property-value pair 1`] = `
Object {
  "@media (max-width: 540px)": Object {
    "width": "100vw",
  },
  "@media (min-width: 1280px)": Object {
    "width": "90.625vw",
  },
  "@media (min-width: 541px) and (max-width: 1279px)": Object {
    "width": "calc(83.78378378378379vw + 87.56756756756761px)",
  },
}
`;
